### PR TITLE
Extracting signing logic in a Project plugin

### PR DESCRIPTION
**Pull request by cdsap**
Issue #472 raised an incompatibility with conf cache when building the project because of the Signing plugin. 

The Signing plugin is not required for local builds(neither in the CI integration tests) but we need to include the plugin for publishing purposes.  The build is not failing because of the conf cache issues but is affecting the parallelization of the tests.

This PR wraps the build logic of applying the Signing plugin in a Project plugin. The logic will check if we are in CI before applying the logic.

I tried to use convention plugins but we can not apply conditionally two different convention plugins. 

## Tests
Build execution `clean build` and populating conf cache:
https://ge.solutions-team.gradle.com/s/gm2sir7ymd3ci/console-log?page=1#L98

Next Build consuming conf cache:
https://ge.solutions-team.gradle.com/s/am5ygnxxq36wm/console-log?page=1#L1

and executed with parallel execution:
Screenshot 2023-03-14 at 6 14 56 PM

### CODE REVIEW

**Review by clayburn — APPROVED** [review]
LGTM, thanks @cdsap!

**Comment by erichaagdev** [comment]
> I tried to use convention plugins but we can not apply conditionally two different convention plugins.

Could you elaborate a bit more on this?

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+plugins {
+    id 'java-gradle-plugin'
+}
+
+gradlePlugin {
+    plugins {
+        signingPlugin {
+            id = 'org.gradle.android.signing'
```
[the remark this answers is not in the document]
Comment (etiennestuder) [reply buildSrc/build.gradle:8]: If this  plugin lives in the buildSrc folder, i.e. bound to this cache-fix project, I would use the same plugin prefix as the main plugin: `org.gradle.android.cache-fix.convention`.

[the remark this answers is not in the document]
Comment (cdsap) [reply buildSrc/build.gradle]: @erichaagdev not sure about using the convention suffix, in theory, Convention plugins are related to precompiled script plugins, in this case, we are defining a binary Project plugin,  but don't have a strong preference for that.  
 `org.gradle.android` is the namespace used by the project. 

@etiennestuder changed to `org.gradle.android.cache-fix.signing` here https://github.com/gradle/android-cache-fix-gradle-plugin/pull/473/commits/8c62e26efad790a11f33a0b561eb9462e39d88b1

**Comment by cdsap** [comment]
> > I tried to use convention plugins but we can not apply conditionally two different convention plugins.
> 
> Could you elaborate a bit more on this?

@erichaagdev because the Signing plugin is not compatible with the scenario described in the issue, we are applying the signing plugin conditionally based on the environment.  
I tested different approaches with convention plugins but we can't apply logic in the plugins block. 
```
plugins {
    if(x){
       id 'signing-convention`
    } else {
      id `signing-convention-no-op`
   }
 }
```
Same applies in the plugins block defined in the convention plugin. That's why a Binary plugin is more flexible at the time to apply the build logic

**Comment by cdsap** [comment]
after checking with @erichaagdev, we will wait for Gradle 8.1 to verify if the Plugin is compatible, if it's required we will reopen the pr

**Pull request closed without merging**